src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberFieldSymbol.cs

```diff
@@ -455,6 +455,11 @@ internal sealed override TypeSymbol GetFieldType(ConsList<FieldSymbol> fieldsBei
                         {
                             diagnosticsForFirstDeclarator.Add(ErrorCode.ERR_ImplicitlyTypedVariableMultipleDeclarator, typeSyntax.Location);
                         }
+                        else if (this.IsConst && this.ContainingType.IsScriptClass)
+                        {
+                            // For const var in script, we won't try to bind the initializer (case below), as it can lead to an unbound recursion
+                            type = null;
+                        }
```

Comment:
Added tests for semantic model. It seems fine. The only problem I can see is that "var" is not inferred (it remains a "var" error type).